galley/pkg/crd/validation/config.go

```diff
@@ -113,6 +113,34 @@ func createOrUpdateWebhookConfigHelper(
 	return false, nil
 }
 
+// Delete validatingwebhookconfiguration if exists to cover the case in which update from enabled validation to disabled validation
+func (wh *Webhook) deleteWebhookConfig() {
+	client := wh.clientset.AdmissionregistrationV1beta1().ValidatingWebhookConfigurations()
+
+	deleted, err := deleteWebhookConfigHelper(client, wh.webhookName)
+	if err != nil {
+		scope.Errorf("%v validatingwebhookconfiguration delete failed: %v", wh.webhookName, err)
+		reportValidationConfigDeleteError(err)
+	}
+	scope.Infof("Delete %v validatingwebhookconfiguration is %v", wh.webhookName, deleted)
+}
+
+// Delete validatingwebhookconfiguration if exists. otherwise, do nothing
+func deleteWebhookConfigHelper(
+	client admissionregistration.ValidatingWebhookConfigurationInterface,
+	webhookName string,
+) (bool, error) {
+	_, err := client.Get(webhookName, metav1.GetOptions{})
+	if err != nil {
```

Comment:
Get+Delete can race if another pod is also trying to create the resource.

Reply:
fixed

Reply:
I think the webhook continuously reconcile the config can resolve this race. If my understanding is not right, please elaborate more how to fix this race issue. Thanks.

Reply:
The cleanup path exits after it successfully deletes the config. If someone re-creates the config, galley doesn't re-delete it. You'll need to decouple overall webhook config reconciliation loop from the webhook http handler itself.

Reply:
@ayj Hi Jason, from my previous commit, I have changed to cleanup to continuously reconcile the config (remove the `return` from cleanup). in fact, w/validation=false, we won’t start the validation server so that the existing webhook config reconciliation logic is not performed. Due to the different time window, I cannot find you from the slack. I hope this fix can be in release 1.1. Please help on it. Thanks.

Reply:
@ayj Hi Jason, I would like to switch to `master` branch for this PR if you think it is risky for release-1.1 right now. comments? Thanks.

Reply:
Switching to master sgtm